src/Lecture3.hs

```diff
@@ -83,7 +83,22 @@ Tuesday
   would work for **any** enumeration type in Haskell (e.g. 'Bool',
   'Ordering') and not just 'Weekday'?
 -}
-next = error "TODO"
+
+class (Eq a,Enum a, Bounded a) => CircleEnum a where
+  cpred :: a -> a
+  cpred x 
+    | x == minBound = maxBound
+    | otherwise = pred x
+
+  csucc :: a -> a
+  csucc x
+    | x == maxBound = minBound
+    | otherwise = succ x
```

Comment:
That's the neat polymorphic implementation! 👏🏻 

In this exercise, you don't need to create an extra typeclass, you can implement `csucc` directly in `next` but it's great to see that you get the concept 🙂

Reply:
Ha, you are right! What a big circle of my implementation. 😆